connectivity/tests/pod.go

```diff
@@ -165,3 +167,81 @@ func (s *podToPodWithEndpoints) curlEndpoints(ctx context.Context, t *check.Test
 		}
 	}
 }
+
+// PodToPodNoFrag is a test to check whether a correct MTU is set
+// for pods. The check is performed by sending an ICMP Echo request with DF
+// set ("do not fragment"). The ICMP payload size of the request:
+//
+// - For IPv4: $POD_MTU - 20 (IPv4 hdr) - 8 (ICMP Echo hdr)
+// - For IPv6: $POD_MTU - 40 (IPv6 hdr) - 8 (ICMP Echo hdr)
+func PodToPodNoFrag() check.Scenario {
+	return &podToPodNoFrag{}
+}
+
+type podToPodNoFrag struct{}
+
+func (s *podToPodNoFrag) Name() string {
+	return "pod-to-pod-no-frag"
+}
+
+func (s *podToPodNoFrag) Run(ctx context.Context, t *check.Test) {
+	ct := t.Context()
+	client := ct.RandomClientPod()
+	var mtu int
+
+	cmd := []string{
+		"/bin/sh", "-c",
+		"ip route show default | grep -oE 'mtu [^ ]*' | cut -d' ' -f2",
+	}
+	t.Debugf("Running %s", strings.Join(cmd, " "))
+	mtuBytes, err := client.K8sClient.ExecInPod(ctx, client.Pod.Namespace,
+		client.Pod.Name, "", cmd)
+	if err != nil {
+		t.Fatalf("Failed to get route MTU in pod %s: %s", client, err)
+	}
+
+	// Derive MTU from pod iface instead
+	if mtuBytes.Len() == 0 {
+		cmd := []string{
```

Comment:
maybe easier to directly `cat /sys/class/net/eth0/mtu` instead of shelling out to `ip`?